sycl/source/detail/device_impl.hpp

```diff
@@ -66,6 +66,107 @@ class device_impl : public std::enable_shared_from_this<device_impl> {
   };
   friend class platform_impl;
 
+  bool has_info_desc(ur_device_info_t Desc) const {
```

Comment:
Had to move this block of code to before first use because of some C++ rules when type deduction takes place.